src/components/OSCALControlImplementationAdd.js

```diff
@@ -0,0 +1,120 @@
+import React, { useRef, useState } from "react";
+import Grid from "@material-ui/core/Grid";
+import { TextField } from "@material-ui/core";
+import AddBoxIcon from "@material-ui/icons/AddBox";
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+import SaveIcon from "@material-ui/icons/Save";
+import { Autocomplete } from "@material-ui/lab";
+import { v4 as uuidv4 } from "uuid";
+import OSCALEditableFieldActions, {
+  getElementLabel,
+} from "./OSCALEditableFieldActions";
+import * as restUtils from "./oscal-utils/OSCALRestUtils";
+
+// calculates possible new controls
+function getControlIdsAndTitles(controls, implementedControls) {
+  const controlIdsAndTitles = {};
+  controls
+    .filter((control) => !implementedControls.includes(control.id))
+    .forEach((control) => {
+      const title = `${control.id.toUpperCase()} ${control.title}`;
+      controlIdsAndTitles[title] = control.id;
+    });
+  return controlIdsAndTitles;
+}
+
+export default function OSCALControlImplementationAdd(props) {
+  const [inEditState, setInEditState] = useState(false);
+  const [isProcessingRequest, setIsProcessingRequest] = useState(false);
+  const [newControl, setNewControl] = useState("");
+  const oldImplementedRequirements = useRef(props.implementedRequirements);
+  const editIcon = <AddBoxIcon />;
```

Comment:
With the current control/component navigation it's not clear what this button will do.

I'd suggest using a text button with "Add Control" text, perhaps using the secondary color.